docs/content/resource-authors-guide.md

```diff
@@ -134,10 +135,39 @@ type Shell struct {
 ### Exporting Values
 
 Converge will automatically extract values from a `resource.Task` that are
-annotated with the `export` or `re-export-as` struct tags.  Fields that are
-exported with `export` will be available with the name that they are exported
-with.  In the example above the fields of `Status` would be available as
-`status.fieldname`.
+annotated with the `export` or `re-export-as` struct tags.  For fields that are
+exported with `export`, they can be referenced directly.  For example if we have
+the following task:
+
+```hcl
+task "foo" {
+  check = "test -f foo.txt"
+  apply = "touch foo.txt"
+}
+```
+
+we may reference any of the exported fields in a lookup by typing `"{{lookup
+'task.foo.field'}}"`; for example `"{{lookup 'task.foo.dir'}}"` or `"{{lookup
+'task.foo.env'}}"`.  Re-exported fields will provide a namespace for structs
+that also export values.  In our `Shell` example we are re-exporting a
+`CommandResults` struct, show below:
+
+```go
+type CommandResults struct {
+    ResultsContext
+    ExitStatus uint32 `export:"exitstatus"`
+    Stdout     string `export:"stdout"`
+    Stderr     string `export:"stderr"`
+    Stdin      string `export:"stdin"`
+    State      *os.ProcessState
+}
+```
+
+since `CommandResults` exports `stdout`, `stderr`, and `stdin` we can reference
+these values, for example: `"{{lookup 'task.foo.status.stdout'}}"` or
+`"{{lookup 'task.foo.status.stderr'}}"`.
+
```

Comment:
I wonder about including an hcl example that uses lookups for export and re-export-as?